pycbc/filter/qtransform.py

```diff
@@ -42,7 +42,160 @@
 from pycbc.types import zeros
 from numpy import fft as npfft
 import os
+import h5py
+from pycbc.filter import highpass_fir, matched_filter
+from pycbc.psd import welch, interpolate
 
+def pycbc_insp_main(segments, filename='q_info.hdf'):
+    """Main function for pycbc_inspiral implementation of qtransform.py
+    Parameters
+    ----------
+    segments:
+        list of pycbc frequency series segments
+    filename:
+        name of output file for qtransform info to be stored
+
+    """
+
+    comb_q_dict = {}
+    for s_num, stilde in enumerate(segments):
+        # getting q-tiles for segments
+        Qbase, q_frange, q_data = pycbc_insp_tiling(stilde, stilde.psd)
+
+        # getting q-plane for segment
+        Qplane, interp, qs_time, qe_time = qplane(Qbase, q_data, q_frange, fres=None, seg=stilde)
+        
+        del q_frange, q_data
+
+        # write q info to an hdf file
+        Qbase_tmp = {}
+        Qplane_tmp = {}
+        print 'start time: %s, end time: %s' % (qs_time, qe_time)
+        if not 'qtiles' in comb_q_dict:
+            Qbase_tmp['seg_%s-%s' % (str(qs_time),str(qe_time))] = Qbase
+            comb_q_dict['qtiles'] = Qbase_tmp
+        if not 'qplanes' in comb_q_dict:
+            Qplane_tmp['seg_%s-%s' % (str(qs_time),str(qe_time))] = Qplane
+            comb_q_dict['qplanes'] = Qplane_tmp
+        else:
+            comb_q_dict['qtiles']['seg_%s-%s' % (str(qs_time),str(qe_time))] = Qbase
+            comb_q_dict['qplanes']['seg_%s-%s' % (str(qs_time),str(qe_time))] = Qplane
+
+    # save qtransform info to hdf5 file
+    #path = '/'
+    #recursively_save_dict_contents_to_group(out_vals, path, comb_q_dict)
+
+    return comb_q_dict
+
+def pycbc_insp_tiling(seg, frange=(0,1024), qrange=(4,64)):
+    """Iterable constructor of QTile tuples
+    Parameters
+    ----------
+    seg:
+        fft'd analysis chunck
+    frange:
+        frequency range
+    qrange:
+        q range
+
+    Returns
+    -------
+    Qbase: 'dict'
+        dictionary containing Q-tile tuples for a set of Q-planes
+    frange: 'list'
+        upper and lower bounds on frequency range
+    data:
+        whitened strain of analysis chunk
+
+    """
```

Comment:
@ahnitz I went ahead and removed the additional highpass filter, removed the recalculation of the psd line, changed a few variable names in the function.